packages/foundry/contracts/utils/ContangoWand.sol

```diff
@@ -0,0 +1,208 @@
+        require(min >= bounds_.min, "Min debt out of bounds");
+
+        cauldron.setDebtLimits(baseId, ilkId, max, min, bounds_.dec);
+    }
+
+    /// ----------------- Oracle Governance -----------------
+
+    /// @notice Set a pool as a source in the YieldSpace oracle, as long as:
+    /// - It is a pool known to the Yield Ladle
+    /// - The baseId matches the pool's baseId
+    /// - The quoteId matches the pool's seriesId
+    function setYieldSpaceOracleSource(bytes6 seriesId) external auth {
+        IPool pool_ = IPool(masterLadle.pools(seriesId));
+        require(address(pool_) != address(0), "Pool not known to the Yield Ladle");
+        DataTypes.Series memory series_ = masterCauldron.series(seriesId);
+        require(address(series_.fyToken) != address(0), "Series not known to the Yield Cauldron");
+        require(address(series_.fyToken) == address(pool_.fyToken()), "fyToken mismatch"); // Sanity check
+
+        yieldSpaceOracle.setSource(series_.baseId, seriesId, pool_);
+    }
+
+    /// @notice Set the YieldSpace oracle as the source for a given asset pair in the Composite oracle, provided the source is set in the YieldSpace oracle
+    function setCompositeOracleSource(bytes6 baseId, bytes6 ilkId) external auth {
+        (IPool pool_, ) = yieldSpaceOracle.sources(baseId, ilkId);
+        require(address(pool_) != address(0), "YieldSpace oracle not set");
+        compositeOracle.setSource(baseId, ilkId, yieldSpaceOracle);
+    }
+
+    /// @notice Set a path in the Composite oracle, as long as the path is not overwriting anything
+    function setCompositeOraclePath(bytes6 baseId, bytes6 quoteId, bytes6[] calldata path) external auth {
+        require(compositeOracle.paths(baseId, quoteId, 0) == bytes6(0), "Path already set"); // We check that the first element in the path is empty
+        compositeOracle.setPath(baseId, quoteId, path);
+    }
+
+    /// ----------------- Ladle Governance -----------------
+
+    /// @notice Propagate a pool to the Ladle from the Yield Ladle
+    function addPool(bytes6 seriesId) external auth {
+        address pool_ = masterLadle.pools(seriesId);
+        require(pool_ != address(0), "Pool not known to the Yield Ladle");
+        ladle.addPool(seriesId, pool_);
+    }
+
+    /// @notice Propagate an integration to the Ladle from the Yield Ladle
+    function addIntegration(address integration) external auth {
+        ladle.addIntegration(integration, masterLadle.integrations(integration));
+    }
+
+    /// @notice Propagate a token to the Ladle from the Yield Ladle
+    function addToken(address token) external auth {
+        ladle.addToken(token, masterLadle.tokens(token));
+    }
+
+    /// @notice Add join to the Ladle.
+    /// @dev These will often be used to hold fyToken, so it doesn't seem possible to put boundaries. However, it seems low risk. Famous last words.
+    function addJoin(bytes6 assetId, address join) external auth {
```

Comment:
so the `assetId` of a `fyToken` always match the `seriesId`, so you could load the series from the `masterCauldron` and check that `address(series.fyToken) == IJoin(join).base()`.
I'd actually make the second param of type `IJoin` to avoid the casting.

in any case, this way you can be sure that the added `Join` is at least meant to hold the right token

Reply:
Also I'm thinking we could also have a factory method that deploys the `Join` and give the appropriate permissions, as Joins for fyTokens are never created as part of Yield's roll